src/transformers/generation/utils.py

```diff
@@ -1861,6 +1861,10 @@ def generate(
             model_kwargs["attention_mask"] = self._prepare_attention_mask_for_generation(
                 inputs_tensor, generation_config._pad_token_tensor, generation_config._eos_token_tensor
             )
+        elif kwargs_has_attention_mask:
+            # TODO (joao): generalize this check with other types of inputs
+            if model_input_name == "input_ids" and len(model_kwargs["attention_mask"].shape) > 2:
```

Comment:
I'm not sure `len(model_kwargs["attention_mask"].shape) > 2` generalizes (e.g. some audio models have 3d inputs), so I'm fixing the most common use case now and adding a TODO in case this issue pops up again for other input formats